py/tests/tests_unit_kmeans.py

```diff
@@ -0,0 +1,101 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Unit tests for PySparkling H2OKMeans
+"""
+import os
+import sys
+
+sys.path.insert(0, sys.argv[1])
+os.environ['PYSPARK_PYTHON'] = sys.executable
+import unittest
+from pysparkling.context import H2OContext
+from pysparkling.conf import H2OConf
+from pysparkling.ml import H2OKMeans
+from pyspark.sql import SparkSession
+import unit_test_utils
+import generic_test_utils
+from pyspark.mllib.linalg import *
+from pyspark.sql.types import *
+from pyspark.ml import Pipeline, PipelineModel
+
+class H2OKMeansTestSuite(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._conf = unit_test_utils.get_default_spark_conf(cls._spark_options_from_params)
+        cls._spark = SparkSession.builder.config(conf=cls._conf).getOrCreate()
+        cls._hc = H2OContext.getOrCreate(cls._spark, H2OConf(cls._spark).set_cluster_size(1))
+        cls.dataset = cls._spark.read.csv("file://" + unit_test_utils.locate("smalldata/iris/iris_wheader.csv"),
+                                           header=True, inferSchema=True)
+
+
+    def test_H2OKMeans_Pipeline_serialization_and_deserialization(self):
```

Comment:
Should I use snake case for TE tests as well?

Reply:
Originally, I was trying to use camelCase just for the publicly facing API to be consistent with Scala, still use snake case for anything else as that is the Python way. But we can be fully consistent across the code-base

Tried to make the test names shorter so they are easier to read in camel case

Reply:
Readability is fine, I'm just wondering what naming conventions I should use myself when writing new code  to SW . Just want to be consistent with you ...

Reply:
It was just my old habit :). Let's use camel case for anything in PySparkling, I will make sure to keep up with that, to be consistent as well